src/mqt/predictor/rl/PredictorEnv.py

```diff
@@ -191,13 +201,20 @@ def apply_action(self, action_index):
                         elem.apply(tket_qc)
                     altered_qc = tk_to_qiskit(tket_qc)
                 except Exception as e:
-                    print("Error in executing TKET transpile pass: ", action_index)
-                    print(e)
+                    self.logger.error(
+                        "Error in executing TKET transpile pass: "
+                        + ", "
+                        + action["name"]
+                        + ", "
+                        + self.state.name
+                        + ", "
+                        + e,
+                    )
                     return False
             else:
                 raise ValueError(f"Origin {action['origin']} not supported.")
         else:
-            print("ERROR: Action not found. Original QC returned.")
+            self.logger.error("Action not found. Original QC returned.")
             altered_qc = self.state
```

Comment:
that should most likely be an exception. If an action is not found, something definitely went wrong and should be investigated.

Reply:
Done.